the-collab-lab/tcl-21-smart-shopping-list
# Tm vs new shopping list

_For an example of how to fill this template out, [see this Pull Request](https://github.com/the-collab-lab/tcl-3-smart-shopping-list/pull/44)._ 

## Description



1. Added a WelcomeScreen and NavBar component to migrate the add-item-view and list-view outside of the home page

2. User is now able to generate a unique token that is stored to local storage.

3. If a token is present the user is redirected to the list associated with that token.

4. If no token is present the user is redirected to the welcome screen and prompted to generate a token.


## Related Issue



## Acceptance Criteria



1. For users who do not already have a token/list, a button or link exists on the home screen that allows them to create a new list

2. Clicking the button/link generates a new token and saves it to localStorage

3. Once the token has been created and saved, the user is redirected to the "list" view

4. For users who do have a token already saved in localStorage, they should be shown/redirected to the "list" view

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

## Updates

### Before




### After




## Testing Steps / QA Criteria



1. Pull down remote branches: git pull
2. Navigate to correct branch: git checkout tm-vs-new-shopping-list
3. Run server and view in browser on localhost:3000: npm start
4. Click "Create a New List" to generate a new token that can be viewed in the Local Storage dropdown under the Applications tab in the browser's developer tools
5. Clicking the "Create a New List" button should then redirect you to the list view
